From eb31017284f9a1c95602a9c06d606df6b558a691 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Thu, 14 Apr 2022 17:39:32 -0700
Subject: [PATCH] ola: fix compilation with musl 1.2.3

musl 1.2.3 defines NULL as nullptr. cannot use reinterpret_cast with
nullptr.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
[Retrieved from:
https://github.com/OpenLightingProject/ola/pull/1773/commits/eb31017284f9a1c95602a9c06d606df6b558a691]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 ola/AutoStart.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ola/AutoStart.cpp b/ola/AutoStart.cpp
index 89fa51f115..4dbe20d317 100644
--- a/ola/AutoStart.cpp
+++ b/ola/AutoStart.cpp
@@ -110,11 +110,11 @@ TCPSocket *ConnectToServer(unsigned short port) {
     // Try to start the server, we pass --daemon (fork into background) and
     // --syslog (log to syslog).
     execlp("olad", "olad", "--daemon", "--syslog",
-#ifdef __FreeBSD__
-           reinterpret_cast<char*>(0));
+#if __cplusplus >= 201103L
+           nullptr);
 #else
            reinterpret_cast<char*>(NULL));
-#endif  // __FreeBSD__
+#endif  // __cplusplus >= 201103L
     OLA_WARN << "Failed to exec: " << strerror(errno);
     _exit(1);
   }
